modyn/trainer_server/internal/dataset/local_dataset_writer.py

```diff
@@ -1,5 +1,8 @@
+from __future__ import annotations
```

Comment:
To allow for the `str | Path` type annotation syntax. As we are using Python 3.11+ already, this is not needed, however, mypy isn't aware of that as we are missing some configurations in the project root. I will probably look into that eventually, but I'm afraid that adding configurations could reveal many other things.

Reply:
haha you might be right. I did not know that mypy does not catch the A | B syntax, but it could also be that I used Union[] in many places because of this.